apps/accounts/validators.py

```diff
@@ -0,0 +1,237 @@
+import datetime
+import re
+import warnings
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.db.models import Q
+from .models import (
+    UserPasswordDescriptor,
+    PastPassword,
+    PasswordHasher,
+)
+
+
+class PasswordComplexityValidator():
+    '''
+    BB2-62 POAM strengthen blue button developer account authentication
+    - increase password complexity
+    '''
+
+    def __init__(
+            self,
+            min_length_digit=1,
+            min_length_alpha=1,
+            min_length_special=1,
+            min_length_lower=1,
+            min_length_upper=1,
+            special_characters="[~!{}@#$%^&*_+\":;()'[]"
+    ):
+        self.min_length_digit = min_length_digit
+        self.min_length_alpha = min_length_alpha
+        self.min_length_special = min_length_special
+        self.min_length_lower = min_length_lower
+        self.min_length_upper = min_length_upper
+        self.special_characters = special_characters
+
+        settings.PASSWORD_RULES[2]['regex'] = self.special_characters
+
+        self.actaul_params = [
```

Comment:
Is this a typo of `self.actual_params` or intended?

Reply:
good catch, will fix it.